File: java/Level 1/콜라츠 추측.java
class Solution {
    public int solution(long num) { // int형 오버플로우 -> long
        int answer = 0;
        
        while(answer < 500) {
            if(num==1) return answer;
            
            if(num%2==0) {
                num /= 2;
            } else {
                num = num*3 + 1;
            }
            answer += 1;
        }

        return -1;
    }
}

File: python/Level 1/콜라츠 추측.py
def solution(num):
    count = 0
    
    while count < 500:
        if num == 1: # 13번 케이스. 비교 순서 명확히
            return count
        
        count += 1

        if num % 2 == 0:
            num /= 2
        else:
            num = num*3 + 1
    
    return -1